src/NUnitConsole/nunit3-console/TestEventHandler.cs

```diff
@@ -94,31 +98,60 @@ private void SuiteFinished(XmlNode testResult)
             if (outputNode != null)
             {
                 if (_displayLabels == "ON" || _displayLabels == "ALL")
-                    WriteTestLabel(suiteName);
+                    WriteLabelLine(suiteName);
 
-                WriteTestOutput(outputNode);
+                WriteOutputLine(outputNode.InnerText);
             }
         }
 
-        private void WriteTestLabel(string name)
+        private void TestOutput(XmlNode outputNode)
         {
-            using (new ColorConsole(ColorStyle.SectionHeader))
-                _outWriter.WriteLine("=> {0}", name);
+            var testName = outputNode.GetAttribute("testname");
+            var stream = outputNode.GetAttribute("stream");
+
+            if (_displayLabels == "ON" && testName != null)
+                WriteLabelLine(testName);
+
+            WriteOutputLine(outputNode.InnerText, stream == "Error" ? ColorStyle.Error : ColorStyle.Output);
         }
 
-        private void WriteTestOutput(XmlNode outputNode)
+        private string _currentLabel;
+
+        private void WriteLabelLine(string label)
         {
-            using (new ColorConsole(ColorStyle.Output))
+            if (label != _currentLabel)
             {
-                _outWriter.Write(outputNode.InnerText);
+                using (new ColorConsole(ColorStyle.SectionHeader))
+                    _outWriter.WriteLine("=> {0}", label);
+
+                _currentLabel = label;
+            }
+        }
+
+        private void WriteOutputLine(string text)
+        {
+            WriteOutputLine(text, ColorStyle.Output);
+        }
+
+        private void WriteOutputLine(string text, ColorStyle color)
+        {
+            using (new ColorConsole(color))
+            {
+                _outWriter.Write(text);
+
                 // Some labels were being shown on the same line as the previous output
-                if (!outputNode.InnerText.EndsWith("\n"))
+                if (!text.EndsWith("\n"))
                 {
                     _outWriter.WriteLine();
                 }
             }
         }
 
+        private void WriteErrorLine(string text)
```

Comment:
Either it's something I didn't finish or the file did not get saved before I committed. I'll check.